wrappers/services.go

```diff
@@ -372,7 +485,7 @@ func StopServices(apps []*snap.AppInfo, reason snap.ServiceStopReason, inter int
 
 		// ensure the service is really stopped on remove regardless
 		// of stop-mode
-		if reason == snap.StopReasonRemove && !app.StopMode.KillAll() {
+		if reason == snap.StopReasonRemove && !app.StopMode.KillAll() && app.DaemonScope == snap.SystemDaemon {
```

Comment:
Is there a TODO that captures stopping of user services?

Reply:
As mentioned in previous comments I believe this block of code is unnecessary for user services, and possibly system services too on modern distros.  Unless the service has `KillMode=none` or sets `FinalKillSignal` to something other than SIGKILL, systemd will have gone through these steps already.  As we don't allow snaps to alter those settings the sysd.Stop() call should be reliable.

From what I've gathered, this was in place to deal with old buggy versions of systemd, which would be from before they supported a user instance of systemd.